src/common.rs

```diff
@@ -992,6 +1071,30 @@ mod tests {
         println!("{} {:?}", ProcessStatus::Parked, ProcessStatus::Idle);
     }
 
+    #[test]
+    fn test_mac_address_from_str_ok() {
+        let mac = MacAddr::from_str("e5:5d:59:e9:6e:b5").unwrap();
+        let mac = mac.data();
+        assert_eq!(mac[0], 0xe5);
+        assert_eq!(mac[1], 0x5d);
+        assert_eq!(mac[2], 0x59);
+        assert_eq!(mac[3], 0xe9);
+        assert_eq!(mac[4], 0x6e);
+        assert_eq!(mac[5], 0xb5);
+    }
+
+    #[test]
+    fn test_mac_address_from_str_fail() {
+        let result = MacAddr::from_str("127.0.0.1");
+        assert!(result.is_err());
+        let result = MacAddr::from_str("");
+        assert!(result.is_err());
+        let result = MacAddr::from_str("any string");
+        assert!(result.is_err());
+        let result = MacAddr::from_str("e5:5d:rr:59:e9:6e:b5");
+        assert!(result.is_err());
+    }
```

Comment:
Please add a `println!("{}", MacAddr { ... })` somewhere to ensure `Display` is implemented on it. Same with `Debug`.